dlt/pipeline/pipeline.py

```diff
@@ -161,13 +161,28 @@ def _wrap(self: "Pipeline", *args: Any, **kwargs: Any) -> Any:
         for name in self._schema_storage.live_schemas:
             # refresh live schemas in storage or import schema path
             self._schema_storage.commit_live_schema(name)
-        rv = f(self, *args, **kwargs)
-        # save modified live schemas
-        for name in self._schema_storage.live_schemas:
-            self._schema_storage.commit_live_schema(name)
-        # refresh list of schemas if any new schemas are added
-        self.schema_names = self._list_schemas_sorted()
-        return rv
+        try:
+            rv = f(self, *args, **kwargs)
+        except Exception:
+            # because we committed live schema before calling f, we may safely
+            # drop all changes in live schemas
+            for name in list(self._schema_storage.live_schemas.keys()):
+                try:
+                    schema = self._schema_storage.load_schema(name)
```

Comment:
I am wondering if we can load schemas in parallel if there are more than N schemas?

Reply:
@sultaniman what is the question exactly?

Reply:
I mean the cases when there will be dozens or >100 of schemas when calling `self._schema_storage.live_schemas.keys()` and was wondering if we should do it in parallel.